### ISSUE TITLE
Wrong node version error + server not initializing for Django files (Djaneiro, MagicPython)

### ISSUE THREAD

**Issue by faaizajaz** [sublimelsp/LSP-pyright#22]
I am trying to get Pyright working on ST3 on Ubuntu 20.04. I have installed the latest node LTS version (12.19.0) using nvm and set it as the default node version, but LSP keeps giving me the "Installed node version (10.19.0) is lower than required version (12.0.0)".

Output of `node --version` is `12.19.0`.

Since I am using nvm, the binary in `/usr/bin` is the is older `10.9.0` version that's available on the Ubuntu repositories. How do I tell LSP/pyright to use the correct node version?

I've tried adding `"node_path"` to my settings to no success.

**Comment by rchl** [comment]
Overriding the path for `LSP-pyright` specifically is an option but ideally, you should expose the NVM's node to ST.

You can do that by including nvm initialization script in `~/.profile` (if using `zsh` you might need to modify `~/.zprofile` instead but try `~/.profile` first).

Remember to restart the system after changing those or at least logging out.

**Comment by faaizajaz** [comment]
Thanks. Your solution worked, though in the end I opted to upgrade the node package in `/usr/bin/`. Whether that is a good idea, we shall see :).

However I still can't get pyright to work. The extension is installed and I am no longer getting any error messages, but the pyright server never initializes. I just see the "LSP initialized" message and nothing else. It is like the server doesn't even attempt to initialize. LSP logging is turned on (but there is nothing in the output) and the pyright server is enabled globally.

Is it possible that it does not initialize because I am trying to use it on a Django project? I also tried adding `source.python.django` to the list of sources in LSP-pyright settings.

Should I close this issue, as my problem is no longer related to the title?

**Comment by rchl** [comment]
See `LSP doesn't start my language server` on https://lsp.readthedocs.io/en/latest/troubleshooting/

Since you are not using the default python syntax, you need to override the `languages` configuration to include the syntax you are using.

(The configuration is simplified in ST4 and doesn't require `syntax` anymore)

**Comment by faaizajaz** [comment]
Serves me right for not reading the docs more closely. I had added the correct scope, but neglected to add the syntax.

I use both the Djaneiro and MagicPython packages (primarily MagicPython, but syntax in some older projects still uses Djaneiro), so in my `LSP-pyright` settings I added to `"syntaxes":{"Packages/MagicPython/grammars/MagicPython.tmLanguage", "Packages/Djaneiro/Syntaxes/Python Django.tmLanguage"}`

Thanks again @rchl, LSP + pyright is just SO GOOD.
